loader/src/test/java/loader/TripleTableLoaderPartByPredTest.java

```diff
@@ -0,0 +1,462 @@
+	}
+
+	/**
+	 * This test verifies an Exception is thrown when an empty file is parsed.
+	 * 
+	 * @throws Exception
+	 */
+	@Test(expected = Exception.class)
+	public void parsingEmptyFile() throws Exception {
+		ClassLoader classLoader = getClass().getClassLoader();
+		File emptyFile = new File(classLoader.getResource("emptyFile.nt").getFile());
+		putFileToHDFS(emptyFile.getAbsolutePath(), "/emptyFile");
+
+		enableSessionForPhysicalPartitioning();
+		spark().sql("DROP DATABASE IF EXISTS emptyFile_db CASCADE");
+		TripleTableLoader tt_loader = new TripleTableLoader("/emptyFile", "emptyFile_db", spark(), false, true, true);
+		tt_loader.load();
+	}
+
+	// Not behaving as expected. Fix after merging (this belongs to a different
+	// feature).
+	public void parsingTriplesWithPrefixes() throws Exception {
+		ClassLoader classLoader = getClass().getClassLoader();
+		File triplesWithPrefixes = new File(classLoader.getResource("triplesWithPrefixes.nt").getFile());
+		putFileToHDFS(triplesWithPrefixes.getAbsolutePath(), "/triplesWithPrefixes");
+
+		enableSessionForPhysicalPartitioning();
+		spark().sql("DROP DATABASE IF EXISTS triplesWithPrefixes_db CASCADE");
+		TripleTableLoader tt_loader = new TripleTableLoader("/triplesWithPrefixes", "triplesWithPrefixes_db", spark(),
+				false, true, true);
+		tt_loader.load();
+
+		// Expected value:
+		TripleBean t1 = new TripleBean();
+		t1.setS("<http://example.org/resource/Res1>");
+		t1.setP("<http://example.org/property/pro1>");
+		t1.setO("<http://example.org/resource/Res:1000>");
+
+		TripleBean t2 = new TripleBean();
+		t2.setS("<http://example.org/resource/Res5>");
+		t2.setP("<http://example.org/property/pro3>");
+		t2.setO("<http://example.org/resource/Res2>");
+
+		ArrayList<TripleBean> triplesList = new ArrayList<TripleBean>();
+		triplesList.add(t1);
+		triplesList.add(t2);
+
+		spark().sql("USE triplesWithPrefixes_db");
+		Dataset<Row> expectedTT = spark().createDataset(triplesList, triplesEncoder).select("s", "p", "o").orderBy("s",
+				"p", "o");
+		Dataset<Row> actualTT = spark().sql("SELECT s,p,o FROM tripletable ORDER BY s,p,o");
+
+		assertDataFrameEquals(expectedTT, actualTT);
+	}
+
+	/**
+	 * Sets the spark session to be able to partition physically.
+	 */
+	public void enableSessionForPhysicalPartitioning() {
+		spark().sql("SET hive.exec.dynamic.partition = true");
+		spark().sql("SET hive.exec.dynamic.partition.mode = nonstrict");
+		spark().sql("SET hive.exec.max.dynamic.partitions = 4000");
+		spark().sql("SET hive.exec.max.dynamic.partitions.pernode = 2000");
+	}
+
+	/**
+	 * This method puts the file in an HDFS folder, so that this can be used for
+	 * applications working on top of Hadoop. In case the HDFS folder passed as
+	 * argument exists, this will be deleted.
+	 * 
+	 * @param localPath
+	 * @param hdfsFolderPath
+	 * @throws IOException
+	 */
+	public void putFileToHDFS(String localPath, String hdfsFolderPath) throws IOException {
```

Comment:
1) The Junit tests requires inheritance from JavaDataFrameSuiteBase to access the provided spark session and context. Morever, using inheritance in JUnit tests is considered a bad practice (just search a bit for that and you will see what I mean).

2) Yes, many methods are repeated. But there is a peculiarity about this library: if you carefully see, these methods use either spark() (session) or jsc() (context) which require that the class has been already fully initialized. So I have to access a method which has also access to the same session/context (inherited from JavaDataFrameSuiteBase). 

I way of fixing this, would be to create an utility class and the add the method in there. But then I should pass the session as argument putFileToHDFS(String localPath, String hdfsFolderPath, JavaSparkSession session). I think this would be the only viable option to avoid inheritance.

Reply:
> The Junit tests requires inheritance from JavaDataFrameSuiteBase to access the provided spark session and context. 
If you have a Base class that extends JavaDataFrameSuiteBase, and your test classes extend Base class. This problem might be solved.

> Morever, using inheritance in JUnit tests is considered a bad practice
Yes, at some cases, I completely agree with that. Maybe a Base class and inheritance are not the best ways to fix this problem.

> would be to create an utility class and the add the method in there. But then I should pass the session as argument putFileToHDFS(String localPath, String hdfsFolderPath, JavaSparkSession session). I think this would be the only viable option to avoid inheritance.

This solution sounds ok to me. You can have something like HDFSWriter class with a method addFile(JavaSparkContext jsc, String localPath, String hdfsFolderPath)

Reply:
I have implemented the solution with the Utility class and it worked. 
Please resolve.